### PR TITLE
GH3671: Detect VS2022 MSBuild when only Build Tools are installed

### LINKED ISSUE #3671: VS2022: msbuild can not be located, only Build Tools are installed
As discussed in https://github.com/cake-build/cake/discussions/3476,
VS2022 BuildTools should be searched in `Program Files (x86)`, too.

### PR DESCRIPTION
**Pull request by augustoproiete**
Closes #3671

### CODE REVIEW

File: src/Cake.Common/Tools/MSBuild/MSBuildResolver.cs [hunk 1]
```diff
@@ -212,32 +212,43 @@ private static DirectoryPath GetVisualStudioPath(ICakeEnvironment environment, M
                 "Enterprise",
                 "Professional",
                 "Community",
-                "BuildTools"
+                "BuildTools",
+                "Preview",
```
Comment (devlead) [inline src/Cake.Common/Tools/MSBuild/MSBuildResolver.cs:216]: Why add back preview? RC removed the preview folder.

Comment (augustoproiete) [reply src/Cake.Common/Tools/MSBuild/MSBuildResolver.cs:216]: The latest Build Tools available for download (`vs_buildtools__1224110841.1636304512.exe`) still use the Preview folder. I installed on my machine first to confirm.

image

It's also on the screenshot of [discussion 3476](https://github.com/cake-build/cake/discussions/3476#discussioncomment-1595161).

Comment (devlead) [reply src/Cake.Common/Tools/MSBuild/MSBuildResolver.cs:216]: I downloaded https://aka.ms/vs/17/release/vs_BuildTools.exe

from https://visualstudio.microsoft.com/downloads/#tools-for-visual-studio-2022-rc-family

image

image

Comment (augustoproiete) [reply src/Cake.Common/Tools/MSBuild/MSBuildResolver.cs:216]: Odd. I downloaded from `https://visualstudio.microsoft.com` as well, but got a different download link - history shows `https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=BuildTools&ch=pre&rel=17`

Do you know if Preview is going to stop working soon and force devs to update to RC or RTM after next week? I'm thinking it doesn't hurt to leave the `Preview` as the last fallback

Comment (devlead) [reply src/Cake.Common/Tools/MSBuild/MSBuildResolver.cs:216]: We've not had previews in released studio versions before with default resolvers before.

**Review by devlead — APPROVED** [review]
LGTM 👍

**Pull request merged**